### DISCUSSION TITLE
Where are pdb files of these  dll ones?

### DISCUSSION THREAD

**Discussion by Wang-Peng-debug** [coin-or/Ipopt#788] [Q&A]
![image](https://github.com/user-attachments/assets/9ef19d71-d927-4c8b-a4df-716f1ec0dbec)
Look! How to generate these .pdb files of its dll ones

**Comment by svigerske** [comment]
The DLLs are build with the `/Z7` option, about which https://learn.microsoft.com/en-us/cpp/build/reference/z7-zi-zi-debug-information-format?view=msvc-170#z7 says
> The /Z7 option produces object files that also contain full symbolic debugging information for use with the debugger. These object files and any libraries built from them can be substantially larger than files that have no debugging information. The symbolic debugging information includes the names and types of variables, functions, and line numbers. No PDB file is produced by the compiler. However, a PDB file can still be generated from these object files or libraries if the linker is passed the /DEBUG option.

I only have a PDB file for the ipopt.exe. Do you want to have that?

**Reply by svigerske** [reply]
We will build with `/Z7`, but I also see now that the debug information does not make it into the DLLs. Also `/Z7` didn't make it to the linker flags for DLL builds (probably libtool took it out). For [3.14.19](https://github.com/coin-or/Ipopt/releases/tag/releases%2F3.14.19), we now build with `LDFLAGS="-Xlinker -Z7"`, which manages to get PDB files build for the DLLs as well.
They are now also included in the `-mdd` archive.
